src/Block/ContainerBlockService.php

```diff
@@ -13,58 +13,90 @@
 
 namespace Sonata\PageBundle\Block;
 
+use Sonata\AdminBundle\Form\FormMapper as AdminFormMapper;
 use Sonata\BlockBundle\Block\BlockContextInterface;
-use Sonata\BlockBundle\Block\Service\BlockServiceInterface;
-use Sonata\BlockBundle\Block\Service\ContainerBlockService as BaseContainerBlockService;
+use Sonata\BlockBundle\Block\Service\AbstractBlockService;
 use Sonata\BlockBundle\Block\Service\EditableBlockService;
 use Sonata\BlockBundle\Form\Mapper\FormMapper;
+use Sonata\BlockBundle\Form\Type\ContainerTemplateType;
 use Sonata\BlockBundle\Meta\Metadata;
 use Sonata\BlockBundle\Meta\MetadataInterface;
 use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\Form\Type\CollectionType;
+use Sonata\Form\Type\ImmutableArrayType;
 use Sonata\Form\Validator\ErrorElement;
+use Symfony\Component\Form\Extension\Core\Type\TextareaType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
  */
-final class ContainerBlockService implements BlockServiceInterface, EditableBlockService
+final class ContainerBlockService extends AbstractBlockService implements EditableBlockService
 {
-    private BaseContainerBlockService $containerBlockService;
-
-    public function __construct(BaseContainerBlockService $containerBlockService)
+    public function configureCreateForm(FormMapper $form, BlockInterface $block): void
     {
-        $this->containerBlockService = $containerBlockService;
+        $this->configureEditForm($form, $block);
     }
 
-    public function execute(BlockContextInterface $blockContext, ?Response $response = null): Response
+    public function configureEditForm(FormMapper $form, BlockInterface $block): void
     {
-        return $this->containerBlockService->execute($blockContext, $response);
-    }
+        if (!$form instanceof AdminFormMapper) {
+            throw new \InvalidArgumentException('This block can only be edited through SonataAdmin');
+        }
 
-    public function load(BlockInterface $block): void
-    {
-        $this->containerBlockService->load($block);
-    }
+        $form->add('settings', ImmutableArrayType::class, [
+            'keys' => [
+                ['code', TextType::class, [
+                    'required' => false,
+                    'label' => 'form.label_code',
+                ]],
+                ['layout', TextareaType::class, [
+                    'label' => 'form.label_layout',
+                ]],
+                ['class', TextType::class, [
+                    'required' => false,
+                    'label' => 'form.label_class',
+                ]],
+                ['template', ContainerTemplateType::class, [
+                    'label' => 'form.label_template',
+                ]],
+            ],
+            'translation_domain' => 'SonataBlockBundle',
```

Comment:
Shouldn't the translation be added to SonataPageBundle then ?

Reply:
Fixed, added translations. (I only added those already present on SonataBlockBundle)